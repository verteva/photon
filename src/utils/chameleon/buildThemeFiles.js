import {
  getChameleonTheme,
  getChameleonThemeNames,
  parseBrandingJsonV2,
} from './';

export const getBuiltThemeFiles = async (files, mergedJson) => {
  const createdFiles = function(key) {
    return mergedJson[key];
  };

  createdFiles.keys = () => files;

  const chameleonThemeNames = getChameleonThemeNames(createdFiles);
  const themes = {
    themes: chameleonThemeNames,
  };
  const promiseMap = chameleonThemeNames.map(async themeName => {
    const chameleonTheme = await getChameleonTheme(createdFiles, themeName);
    const parsedTheme = parseBrandingJsonV2(chameleonTheme);
    return {
      [themeName]: parsedTheme,
    };
  }, {});

  const resolvedMap = await Promise.all(promiseMap);

  const chameleonThemes = resolvedMap.reduce(
    (acc, item) => ({ ...acc, ...item }),
    {}
  );

  return {
    themes,
    ...chameleonThemes,
  };
};
