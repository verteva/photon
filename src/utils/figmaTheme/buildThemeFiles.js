import { getFigmaTheme, getFigmaThemeNames, parseFigmaJson } from './';

export const getBuiltThemeFiles = async (files, mergedJson) => {
  const createdFiles = function(key) {
    return mergedJson[key];
  };

  createdFiles.keys = () => files;

  const figmaThemeNames = getFigmaThemeNames(createdFiles);
  const themes = {
    themes: figmaThemeNames,
  };
  const promiseMap = figmaThemeNames.map(async themeName => {
    const figmaTheme = await getFigmaTheme(createdFiles, themeName);
    const parsedTheme = parseFigmaJson(figmaTheme);
    return {
      [themeName]: parsedTheme,
    };
  }, {});

  const resolvedMap = await Promise.all(promiseMap);

  const figmaThemes = resolvedMap.reduce(
    (acc, item) => ({ ...acc, ...item }),
    {}
  );

  return {
    themes,
    ...figmaThemes,
  };
};
