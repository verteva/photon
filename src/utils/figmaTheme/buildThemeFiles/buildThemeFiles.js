import { getFigmaTheme, getFigmaThemeNames, parseFigmaJson } from '../';

export const getBuiltThemeFiles = (files, mergedJson) => {
  const createdFiles = function(key) {
    return mergedJson[key];
  };

  createdFiles.keys = () => files;

  const figmaThemeNames = getFigmaThemeNames(createdFiles);
  const themes = {
    themes: figmaThemeNames,
  };
  const figmaThemesMap = figmaThemeNames.map(themeName => {
    const figmaTheme = getFigmaTheme(createdFiles, themeName);
    const parsedTheme = parseFigmaJson(figmaTheme);
    return {
      [themeName]: parsedTheme,
    };
  }, {});

  const figmaThemes = figmaThemesMap.reduce(
    (acc, item) => ({ ...acc, ...item }),
    {}
  );

  return {
    themes,
    ...figmaThemes,
  };
};
