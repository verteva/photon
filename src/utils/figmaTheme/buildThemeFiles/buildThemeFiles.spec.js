import { getBuiltThemeFiles } from './buildThemeFiles';
import {
  correctFiles,
  correctNestedFiles,
  correctSourceFiles,
} from '../mockThemeFiles/testFileObjects';

describe('figmaTheme/buildThemeFiles', () => {
  it('builds theme files for themes', () => {
    const fileNames = Object.keys(correctFiles);
    const res = getBuiltThemeFiles(fileNames, correctFiles);
    expect(res).toMatchSnapshot();
  });

  it('builds theme files for nested themes', () => {
    const fileNames = Object.keys(correctNestedFiles);
    const res = getBuiltThemeFiles(fileNames, correctNestedFiles);
    expect(res).toMatchSnapshot();
  });

  it('builds theme files for source themes', () => {
    const fileNames = Object.keys(correctSourceFiles);
    const res = getBuiltThemeFiles(fileNames, correctSourceFiles);
    expect(res).toMatchSnapshot();
  });
});
