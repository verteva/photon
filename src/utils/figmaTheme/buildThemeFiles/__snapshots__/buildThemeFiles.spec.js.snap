// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`figmaTheme/buildThemeFiles builds theme files for nested themes 1`] = `
Object {
  "theme1": Object {
    "colors": Object {
      "black": "#000000",
      "brand": Object {
        "green": "#219653",
        "orange": "orange",
        "purple": "#6f53f9",
        "red": "#eb5757",
        "teal": "#31839b",
      },
      "grey": Object {
        "00": "#07131f",
        "01": "#6a6e80",
        "02": "#949494",
        "03": "#b5b5b5",
        "08": "#f4f4f4",
        "09": "grey",
        "10": "#f9f9f9",
      },
      "theme": Object {
        "primary": "",
        "secondary": "#31839b",
        "tertiary": "#eb5757",
      },
      "white": "#ffffff",
    },
    "spacing": Object {
      "default": "2",
    },
  },
  "theme2": Object {
    "colors": Object {
      "black": "#000000",
      "brand": Object {
        "green": "#219653",
        "orange": "purple",
        "purple": "#6f53f9",
        "red": "#eb5757",
        "teal": "#31839b",
      },
      "grey": Object {
        "00": "#07131f",
        "01": "#6a6e80",
        "02": "#949494",
        "03": "#b5b5b5",
        "08": "#f4f4f4",
        "09": "#f6f6f6",
        "10": "#f9f9f9",
      },
      "theme": Object {
        "primary": "",
        "secondary": "#31839b",
        "tertiary": "#eb5757",
      },
      "white": "#ffffff",
    },
    "spacing": Object {
      "default": "2",
    },
  },
  "themes": Object {
    "themes": Array [
      "theme1",
      "theme2",
    ],
  },
}
`;

exports[`figmaTheme/buildThemeFiles builds theme files for source themes 1`] = `
Object {
  "theme1": Object {
    "colors": Object {
      "brand": Object {
        "orange": "orange",
      },
      "grey": Object {
        "09": "grey",
      },
      "theme": Object {
        "primary": "",
      },
    },
  },
  "theme2": Object {
    "colors": Object {
      "brand": Object {
        "orange": "purple",
      },
      "theme": Object {
        "primary": "",
      },
    },
  },
  "themes": Object {
    "themes": Array [
      "theme1",
      "theme2",
    ],
  },
}
`;

exports[`figmaTheme/buildThemeFiles builds theme files for themes 1`] = `
Object {
  "theme1": Object {
    "colors": Object {
      "black": "#000000",
      "brand": Object {
        "green": "#219653",
        "orange": "orange",
        "purple": "#6f53f9",
        "red": "#eb5757",
        "teal": "#31839b",
      },
      "grey": Object {
        "00": "#07131f",
        "01": "#6a6e80",
        "02": "#949494",
        "03": "#b5b5b5",
        "08": "#f4f4f4",
        "09": "grey",
        "10": "#f9f9f9",
      },
      "theme": Object {
        "primary": "",
        "secondary": "#31839b",
        "tertiary": "#eb5757",
      },
      "white": "#ffffff",
    },
  },
  "theme2": Object {
    "colors": Object {
      "black": "#000000",
      "brand": Object {
        "green": "#219653",
        "orange": "purple",
        "purple": "#6f53f9",
        "red": "#eb5757",
        "teal": "#31839b",
      },
      "grey": Object {
        "00": "#07131f",
        "01": "#6a6e80",
        "02": "#949494",
        "03": "#b5b5b5",
        "08": "#f4f4f4",
        "09": "#f6f6f6",
        "10": "#f9f9f9",
      },
      "theme": Object {
        "primary": "",
        "secondary": "#31839b",
        "tertiary": "#eb5757",
      },
      "white": "#ffffff",
    },
  },
  "themes": Object {
    "themes": Array [
      "theme1",
      "theme2",
    ],
  },
}
`;
